<launch>

  <node name="kuka_lwr_safety_node" pkg="safe_robot" type="safe_robot_node.py" output="screen">
    <param name="exotica_xml_filename" type="str" value="$(find safe_robot)/configs/kuka_lwr/kuka_lwr.xml"/>

    <!-- Joint position limit checker -->
    <param name="checker/jointpositionlimit/active" type="bool" value="true"/>


    <!-- Joint velocity limit checker -->
    <param name="checker/jointvelocitylimit/active" type="bool" value="true"/>
    <param name="joint_velocity_limit" type="str" value="0.174533 0.174533 0.174533 0.174533 0.174533 0.174533 0.174533"/>
    <!-- 0.0872665 rad = 5 deg-->
    <!-- 0.174533 rad = 10 deg-->

    <!-- Link box limit checker -->
    <param name="checker/linkboxlimit/active" type="bool" value="true"/>
    <param name="links" type="str" value="lwr_arm_7_link"/>
    <param name="box_xlim" type="str" value="-0.5 0.5"/>
    <param name="box_ylim" type="str" value="-0.5 0.5"/>
    <param name="box_ylim" type="str" value="-0.5 2.0"/>

    <!-- State validity checker -->
    <param name="checker/statevalidity/active" type="bool" value="true"/>
    <param name="safe_distance" type="double" value="0.0"/>

  </node>

</launch>
